subsys/net/lib/zperf/zperf_shell.c

```diff
@@ -974,30 +1035,79 @@ void zperf_tcp_started(void)
 	tcp_running = true;
 }
 
+static void tcp_session_cb(enum zperf_status status,
+			   struct zperf_results *result)
+{
+	if (zperf_sh == NULL) {
```

Comment:
I just wonder if we should have extra option in the callback (user_data) where the caller supplied data could be given to the callback. And then have the zperf_sh in that user_data var.iable. This has been user pattern in other parts of the code when using callbacks.

Reply:
Good point, I've reworked the code to provide `user_data` pointer with the callback, the shell submodule no longer needs global variable to store the shell instance.